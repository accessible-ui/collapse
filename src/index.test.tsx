/* jest */
import * as React from 'react'
import {renderHook} from '@testing-library/react-hooks'
import {render, fireEvent} from '@testing-library/react'
import {
  Disclosure,
  Trigger,
  Target,
  Close,
  useControls,
  useIsOpen,
  useDisclosure,
} from './index'

describe('<Disclosure>', () => {
  it('should have a custom id', () => {
    const result = render(
      <Disclosure id="foobar">
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot()
  })

  it('should provide context to function child', () => {
    let cxt

    render(
      <Disclosure>
        {context => {
          cxt = context
          return <div />
        }}
      </Disclosure>
    )

    expect(cxt).toMatchSnapshot()
  })

  it('should invoke onChange callback when open state changes', () => {
    const handleChange = jest.fn()

    const {getByTestId} = render(
      <Disclosure onChange={handleChange}>
        <Trigger>
          <button data-testid="btn">open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(handleChange).not.toBeCalled()
    fireEvent.mouseDown(getByTestId('btn'))
    fireEvent.click(getByTestId('btn'))
    expect(handleChange).toBeCalledWith(true)
    fireEvent.mouseDown(getByTestId('btn'))
    fireEvent.click(getByTestId('btn'))
    expect(handleChange).toBeCalledWith(false)
  })
})

describe('<Target>', () => {
  it('should open and close on Trigger click', () => {
    const result = render(
      <Disclosure>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot('closed initially')
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('closed')
  })

  it('should close on escape key', () => {
    const result = render(
      <Disclosure>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot('closed initially')
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
    fireEvent.keyDown(result.getByText('Hello world'), {
      key: 'Escape',
      which: 27,
    })
    expect(result.asFragment()).toMatchSnapshot('closed')
  })

  it(`shouldn't close on escape key if prop is false`, () => {
    const result = render(
      <Disclosure>
        <Target closeOnEscape={false}>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot('closed initially')
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
    fireEvent.keyDown(result.getByText('Hello world'), {
      key: 'Escape',
      code: 27,
    })
    expect(result.asFragment()).toMatchSnapshot('still open')
  })

  it(`should assign to custom styles when opened or closed`, () => {
    const result = render(
      <Disclosure>
        <Target>
          <div style={{fontSize: '2rem'}}>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it(`should apply custom classname when opened or closed`, () => {
    const result = render(
      <Disclosure>
        <Target>
          <div className="custom">Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it(`should apply user defined openClass and closedClass`, () => {
    const result = render(
      <Disclosure>
        <Target closedClass="closed" openClass="open">
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it(`should apply user defined openStyle and closedStyle`, () => {
    const result = render(
      <Disclosure>
        <Target closedStyle={{display: 'none'}} openStyle={{display: 'block'}}>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it(`should be initially open when defined as such`, () => {
    const result = render(
      <Disclosure defaultOpen>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot('initially open')
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('closed')
  })

  it(`should act like a controlled component when 'open' prop is specified`, () => {
    const result = render(
      <Disclosure open>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot('initially open')
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('still open')

    result.rerender(
      <Disclosure open={false}>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot('closed')
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('still closed')
  })

  it('should render into a portal by default ID', () => {
    const portalRoot = document.createElement('div')
    portalRoot.setAttribute('id', 'portals')
    document.body.appendChild(portalRoot)

    const result = render(
      <Disclosure open>
        <Target portal>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.baseElement).toMatchSnapshot()
    document.body.removeChild(portalRoot)
  })

  it('should render into a portal by custom selector object', () => {
    const portalRoot = document.createElement('div')
    portalRoot.setAttribute('class', 'portals')
    document.body.appendChild(portalRoot)

    const result = render(
      <Disclosure open>
        <Target portal={{container: '.portals'}}>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.baseElement).toMatchSnapshot()
    document.body.removeChild(portalRoot)
  })

  it('should render into a portal by custom selector', () => {
    const portalRoot = document.createElement('div')
    portalRoot.setAttribute('class', 'portals')
    document.body.appendChild(portalRoot)

    const result = render(
      <Disclosure open>
        <Target portal=".portals">
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.baseElement).toMatchSnapshot()
    document.body.removeChild(portalRoot)
  })
})

describe('<Trigger>', () => {
  it('should have openClass and closedClass', () => {
    const result = render(
      <Disclosure>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger closedClass="closed" openClass="open">
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it('should have openStyle and closedStyle', () => {
    const result = render(
      <Disclosure>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger closedStyle={{display: 'none'}} openStyle={{display: 'block'}}>
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it('should fire user-defined onClick handler', () => {
    const cb = jest.fn()
    const result = render(
      <Disclosure>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger closedStyle={{display: 'none'}} openStyle={{display: 'block'}}>
          <button onClick={cb}>open me</button>
        </Trigger>
      </Disclosure>
    )

    fireEvent.click(result.getByText('open me'))
    expect(cb).toBeCalledTimes(0)
    fireEvent.mouseDown(result.getByText('open me'))
    fireEvent.click(result.getByText('open me'))
    expect(cb).toBeCalledTimes(1)
  })
})

describe('<Close>', () => {
  it('should close the modal', () => {
    const result = render(
      <Disclosure defaultOpen={true}>
        <Target>
          <div>
            <Close>
              <button data-testid="close">Close me</button>
            </Close>
            Hello world
          </div>
        </Target>

        <Trigger closedClass="closed" openClass="open">
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.mouseDown(result.getByTestId('close'))
    fireEvent.click(result.getByTestId('close'))
    expect(result.asFragment()).toMatchSnapshot('closed')
  })

  it('should fire user-defined onClick handler', () => {
    const cb = jest.fn()
    const result = render(
      <Disclosure defaultOpen={true}>
        <Target>
          <div>
            <Close>
              <button onClick={cb} data-testid="close">
                Close me
              </button>
            </Close>
            Hello world
          </div>
        </Target>

        <Trigger closedClass="closed" openClass="open">
          <button>open me</button>
        </Trigger>
      </Disclosure>
    )

    fireEvent.click(result.getByTestId('close'))
    expect(cb).toBeCalledTimes(0)
    fireEvent.mouseDown(result.getByTestId('close'))
    fireEvent.click(result.getByTestId('close'))
    expect(cb).toBeCalledTimes(1)
  })
})

describe('useControls()', () => {
  it('should have toggle, open, close keys', () => {
    const {result} = renderHook(() => useControls(), {wrapper: Disclosure})
    expect(Object.keys(result.current)).toStrictEqual([
      'open',
      'close',
      'toggle',
    ])
  })
})

describe('useIsOpen()', () => {
  it('should return boolean', () => {
    const {result} = renderHook(() => useIsOpen(), {wrapper: Disclosure})
    expect(typeof result.current).toBe('boolean')
  })
})

describe('useDisclosure()', () => {
  it('should return context', () => {
    const {result} = renderHook(() => useDisclosure(), {wrapper: Disclosure})
    expect(result.current).toMatchSnapshot()
  })
})
