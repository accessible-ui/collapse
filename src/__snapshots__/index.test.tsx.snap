// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Close> should close the modal 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°19"
    style="visibility: visible;"
  >
    <button
      aria-controls="ðŸ…°19"
      aria-expanded="true"
      aria-label="Close"
      data-testid="close"
      role="button"
      tabindex="0"
    >
      Close me
    </button>
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°19"
    aria-expanded="true"
    class="open"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Close> should close the modal: closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°19"
    style="visibility: hidden;"
  >
    <button
      aria-controls="ðŸ…°19"
      aria-expanded="false"
      aria-label="Close"
      data-testid="close"
      role="button"
      tabindex="0"
    >
      Close me
    </button>
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°19"
    aria-expanded="false"
    class="closed"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Close> should override the aria label: aria-label="close me" 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°18"
    style="visibility: visible;"
  >
    <button
      aria-controls="ðŸ…°18"
      aria-expanded="true"
      aria-label="close me"
      data-testid="close"
      role="button"
      tabindex="0"
    >
      Close me
    </button>
    Hello world
  </div>
</DocumentFragment>
`;

exports[`<Disclosure> should have a custom id 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="foobar"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="foobar"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Disclosure> should provide context to function child 1`] = `
Object {
  "close": [Function],
  "id": "ðŸ…°1",
  "isOpen": false,
  "open": [Function],
  "toggle": [Function],
}
`;

exports[`<Target> should act like a controlled component when 'open' prop is specified: closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°11"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°11"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should act like a controlled component when 'open' prop is specified: initially open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°11"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°11"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should act like a controlled component when 'open' prop is specified: still closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°11"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°11"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should act like a controlled component when 'open' prop is specified: still open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°11"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°11"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply custom classname when opened or closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="custom"
    id="ðŸ…°7"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°7"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply custom classname when opened or closed: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="custom"
    id="ðŸ…°7"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°7"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply user defined openClass and closedClass 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="closed"
    id="ðŸ…°8"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°8"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply user defined openClass and closedClass: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="open"
    id="ðŸ…°8"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°8"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply user defined openStyle and closedStyle 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°9"
    style="visibility: hidden; display: none;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°9"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply user defined openStyle and closedStyle: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°9"
    style="visibility: visible; display: block;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°9"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should assign to custom styles when opened or closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°6"
    style="visibility: hidden; font-size: 2rem;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°6"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should assign to custom styles when opened or closed: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°6"
    style="visibility: visible; font-size: 2rem;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°6"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should be initially open when defined as such: closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°10"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°10"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should be initially open when defined as such: initially open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°10"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°10"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should close on escape key: closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°4"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°4"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should close on escape key: closed initially 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°4"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°4"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should close on escape key: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°4"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°4"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should open and close on Trigger click: closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°3"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°3"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should open and close on Trigger click: closed initially 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°3"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°3"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should open and close on Trigger click: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°3"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°3"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should render into a portal by custom selector 1`] = `
<body>
  <div
    class="portals"
  >
    <div
      aria-hidden="false"
      id="ðŸ…°14"
      style="visibility: visible;"
    >
      Hello world
    </div>
  </div>
  <div>
    <button
      aria-controls="ðŸ…°14"
      aria-expanded="true"
      role="button"
      tabindex="0"
    >
      open me
    </button>
  </div>
</body>
`;

exports[`<Target> should render into a portal by custom selector object 1`] = `
<body>
  <div
    class="portals"
  >
    <div
      aria-hidden="false"
      id="ðŸ…°13"
      style="visibility: visible;"
    >
      Hello world
    </div>
  </div>
  <div>
    <button
      aria-controls="ðŸ…°13"
      aria-expanded="true"
      role="button"
      tabindex="0"
    >
      open me
    </button>
  </div>
</body>
`;

exports[`<Target> should render into a portal by default ID 1`] = `
<body>
  <div
    id="portals"
  >
    <div
      aria-hidden="false"
      id="ðŸ…°12"
      style="visibility: visible;"
    >
      Hello world
    </div>
  </div>
  <div>
    <button
      aria-controls="ðŸ…°12"
      aria-expanded="true"
      role="button"
      tabindex="0"
    >
      open me
    </button>
  </div>
</body>
`;

exports[`<Target> shouldn't close on escape key if prop is false: closed initially 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°5"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°5"
    aria-expanded="false"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> shouldn't close on escape key if prop is false: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°5"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°5"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> shouldn't close on escape key if prop is false: still open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°5"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°5"
    aria-expanded="true"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Trigger> should have openClass and closedClass 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°15"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°15"
    aria-expanded="false"
    class="closed"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Trigger> should have openClass and closedClass: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°15"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°15"
    aria-expanded="true"
    class="open"
    role="button"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Trigger> should have openStyle and closedStyle 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    id="ðŸ…°16"
    style="visibility: hidden;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°16"
    aria-expanded="false"
    role="button"
    style="display: none;"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Trigger> should have openStyle and closedStyle: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    id="ðŸ…°16"
    style="visibility: visible;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°16"
    aria-expanded="true"
    role="button"
    style="display: block;"
    tabindex="0"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`useDisclosure() should return context 1`] = `
Object {
  "close": [Function],
  "id": "ðŸ…°23",
  "isOpen": false,
  "open": [Function],
  "toggle": [Function],
}
`;
